### PR TITLE
Feature 11 device authentication

### LINKED ISSUE #11: Device authentication

### PR DESCRIPTION
**Pull request by villejuutila**
Resolves #11

### CODE REVIEW

File: lib/database/dao/keys_dao.dart [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import 'package:drift/drift.dart';
+import 'package:oss_surveys_customer/database/database.dart';
+
+import '../model/key.dart';
```
Comment (EVP1989) [inline lib/database/dao/keys_dao.dart]:
```suggestion
import "../model/key.dart";
```

File: lib/screens/default_screen.dart [hunk 2]
```diff
@@ -1,28 +1,66 @@
...
-        child: SvgPicture.asset(
-              "assets/logo.svg",
-              width: MediaQuery.of(context).size.width * 0.7,
-              height: MediaQuery.of(context).size.height * 0.7
-            ),
+        child: Column(
+          mainAxisAlignment: MainAxisAlignment.center,
+          children: [
+            if (!_isApprovedDevice) const Text(
+              "Laitetta ei ole vielä otettu käyttöön.",
```
Comment (EVP1989) [inline lib/screens/default_screen.dart]: Should we use localized strings?

**Review by EVP1989 — APPROVED** [review]
Approved with some comments

File: lib/main.dart [hunk 3]
```diff
@@ -1,28 +1,98 @@
...
 
+/// Setups MQTT Listeners
+void _setupMqttListeners() {
+  SurveysListener();
+}
+
+/// Setups timers for background tasks ran on interval.
+void _setupTimers() async {
+  if (!isDeviceApproved) {
+    Timer.periodic(const Duration(seconds: 30), (timer) => _pollDeviceApprovalStatus(timer) );
```
Comment (lauriLukkarinen) [inline lib/main.dart]: These kinds of things should be automatically corrected on file save via Dart formatter, are you using Flutter extension in VS Code and is the formatter enabled?
```suggestion
    Timer.periodic(const Duration(seconds: 30), (timer) => _pollDeviceApprovalStatus(timer));
```

File: lib/mqtt/model/status_message.dart [hunk 4]
```diff
@@ -1,21 +1,24 @@
...
-  bool status;
+  bool online;
   
   /// Converts JSON to Status Message
   factory StatusMessage.fromJson(Map<String, dynamic> json) {
-    final status = json["status"] as bool;
+    final onlin = json["online"] as bool;
     
-    return StatusMessage(status);
+    return StatusMessage(onlin);
```
Comment (lauriLukkarinen) [inline lib/mqtt/model/status_message.dart]: online*

File: lib/screens/default_screen.dart (hunk 2 above)
Comment (lauriLukkarinen) [reply lib/screens/default_screen.dart]: We should, there are packages for that.

**Review by lauriLukkarinen — CHANGES_REQUESTED** [review]

File: lib/database/dao/keys_dao.dart (hunk 1 above)
Comment (villejuutila) [reply lib/database/dao/keys_dao.dart]: Resolved

File: lib/main.dart (hunk 3 above)
Comment (villejuutila) [reply lib/main.dart]: Resolved

File: lib/mqtt/model/status_message.dart (hunk 4 above)
Comment (villejuutila) [reply lib/mqtt/model/status_message.dart]: Resolved

File: lib/screens/default_screen.dart (hunk 2 above)
Comment (villejuutila) [reply lib/screens/default_screen.dart]: Resolved

File: lib/l10n/app_fi.arb [hunk 5]
```diff
@@ -0,0 +1,3 @@
+{
+  "notYetApproved": "Laitetta ei ole vielä otettu käyttöön."
```
Comment (lauriLukkarinen) [inline lib/l10n/app_fi.arb:2]: Not the same as app_en.arb?

**Review by lauriLukkarinen — CHANGES_REQUESTED** [review]

File: lib/l10n/app_fi.arb (hunk 5 above)
Comment (villejuutila) [reply lib/l10n/app_fi.arb:2]: English localization is considered as the default localization and according to Flutter documentation, the description (missing in this file but present in app_en.arb) is included in the default locale.
https://docs.flutter.dev/development/accessibility-and-localization/internationalization#adding-your-own-localized-messages

**Review by lauriLukkarinen — APPROVED** [review]

**Pull request merged**